From f952fd99fc1b6934708ae64cbe92ec1c1ec4e419 Mon Sep 17 00:00:00 2001
From: Jerome Benoit <quartermaster@rezozer.net>
Date: Sun, 3 Jul 2022 17:11:31 +0200
Subject: [PATCH] fplll: fix namespace issue

Description: fix fplll namespace issue
 Upstream maintainer of FPLLL has removed from `root` fplll.h
 'using namespace fplll' (commit e886a8899f71ca78711fdf420e6817432b2fe4be),
 this patch mimics the former behviour at the scale of this package.
Origin: vendor, Debian
Author: Jerome Benoit <calculus@rezozer.net>
Last-Update: 2022-07-03
---
 m4/ac_check_fplll.m4 | 5 ++++-
 src/fplll.C          | 2 ++
 2 files changed, 6 insertions(+), 1 deletion(-)

diff --git a/m4/ac_check_fplll.m4 b/m4/ac_check_fplll.m4
index a6e2945..224f87c 100644
--- a/m4/ac_check_fplll.m4
+++ b/m4/ac_check_fplll.m4
@@ -68,7 +68,10 @@ if test "$found_fplll" = false; then
     AX_CXX_COMPILE_STDCXX([11],[noext],[mandatory])
     AC_CHECK_HEADER(fplll.h,[found_fplll=true],[found_fplll=false],[#include <mpfr.h>])
     AC_MSG_CHECKING([for lllReduction in -fplll (version 5.x)])
-    AC_LINK_IFELSE([AC_LANG_PROGRAM([#include <fplll.h>],[ZZ_mat<mpz_t> M(3,3);lll_reduction(M, 0.99, 0.51, LM_WRAPPER);])],[AC_MSG_RESULT([yes]);found_fplll=5],[AC_MSG_RESULT([no]);found_fplll=false])
+    AC_LINK_IFELSE([AC_LANG_PROGRAM([[
+#include <fplll.h>
+using namespace fplll;
+]],[ZZ_mat<mpz_t> M(3,3);lll_reduction(M, 0.99, 0.51, LM_WRAPPER);])],[AC_MSG_RESULT([yes]);found_fplll=5],[AC_MSG_RESULT([no]);found_fplll=false])
 fi
 AC_LANG_POP([C++])
 
diff --git a/src/fplll.C b/src/fplll.C
index 45241c2..832c7b1 100644
--- a/src/fplll.C
+++ b/src/fplll.C
@@ -23,6 +23,8 @@
 #include "floattypes.h"
 
 #include <fplll.h>
+using namespace fplll;
+
 
 typedef Obj (*ObjFunc)(); // I never could get the () and * right
 
-- 
2.38.1

